src/smartcontracts/codec/binary.spec.ts

```diff
@@ -129,6 +130,27 @@ describe("test binary codec (advanced)", () => {
         assert.deepEqual(decodedTopLevel, list);
     });
 
+    it("should fail with large lists using default constraints", async () => {
+        let numItems = 2 ** 17;
+        let codec = new BinaryCodec();
+
+        let items: TypedValue[] = [];
+
+        for (let i = 0; i < numItems; i++) {
+            items.push(new U32Value(i));
+        }
+
+        let list = new List(new ListType(new U32Type()), items);
+
+        assert.throws(() => {
+            codec.encodeNested(list);
+        }, errors.ErrCodec);
```

Comment:
Error message could also have been used in the `assert.throws()`. Here, and below.

Reply:
Added error message.